### PR TITLE
Setup automatic PR assignment

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

Fixes https://github.com/crossplane/crossplane/issues/3046

We already have OWNERS.md, but this format is meaningful to GitHub. The goal of this file is for most PRs to automatically and fairly have one maintainer and two reviewers set as PR reviewers. All maintainers have permission to approve and merge PRs, but reviewers do not. Most PRs should by members of the reviewers group before being passed to a maintainer for final review. This in part depends on how the groups in this file are configured.

 @crossplane/steering-committee - Assigns 3 members. Admin perms to this repo.
 @crossplane/crossplane-maintainers - Assigns 1 member. Maintain perms to this repo.
 @crossplane/crossplane-reviewers - Assigns 2 members. Triage perms to this repo.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: CODEOWNERS
Comment (negz) [inline CODEOWNERS:25]: That's a shame. Apparently triage isn't enough. It seems like I would have to grant @crossplane/crossplane-reviewers write access, which I believe would allow them to approve and merge PRs without approval from a member of @crossplane/crossplane-maintainers .

Comment (negz) [reply CODEOWNERS:25]: Screen Shot 2022-08-12 at 7 31 39 PM

Hrm, it could be that our branch protection would prevent folks without "maintain" or higher permissions _merging_ a PR?

Comment (negz) [reply CODEOWNERS:25]: https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization indicates that write access would typically give folks the ability to merge PRs.

I'll switch the Crossplane-reviewers team over to 'write' on Monday and ask a volunteer in that team to let me know what permissions they have. I'd personally feel comfortable with @crossplane/crossplane-reviewers having permission to approve PRs as long as only @crossplane/crossplane-maintainers can merge them.

Comment (negz) [reply CODEOWNERS:25]: I've given @crossplane/crossplane-reviewers write permissions to test this.

File: CODEOWNERS
Comment (jbw976) [inline CODEOWNERS:16]: the other too are `crossplane/...` but this one is `crossplane-...` - typo?

File: CODEOWNERS
Comment (jbw976) [reply CODEOWNERS:25]: > That's a shame.

not sure if i caught what exactly the shame was here - was there a message or github status or something this was referring to? i may have picked up the context from the rest of your thread though...

yes, our branch protection looks like folks cannot merge PRs without being at least in the maintainer role (write access would not be enough to merge a PR).

the general goal of crossplane-reviewers having permission to APPROVE, but then crossplane-maintainers can MERGE them sounds exactly what we want.

**Review by jbw976 — APPROVED** [review]
looks like a good start to begin iterating on some of these settings as needed! let me know if you need any other eyes on settings/config :)

File: CODEOWNERS
Comment (negz) [reply CODEOWNERS:25]: > not sure if i caught what exactly the shame was here - was there a message or github status or something this was referring to?

Yeah, it seems like GitHub has a built in CODEOWNERS linter that was telling me that @crossplane/crossplane-reviewers had insufficient permissions to be a code owner. That went away automatically when I bumped them from triage to write.

File: CODEOWNERS
Comment (negz) [reply CODEOWNERS:16]: Aha good catch. I had remembered there was something I wanted to fix in here but not what it was.

**Pull request merged**
